### PR TITLE
Improve test coverage

### LINKED ISSUE #36: Improve test coverage
Currently the test coverage is at 55 % – this is not really good for a package called `zope.testing`.

### PR DESCRIPTION
**Pull request by icemac**
Show deprecation warnings when importing modules which are not ported to Python 3.

Fixes #36.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -4,7 +4,8 @@ Changes
 4.10 (unreleased)
 -----------------
 
-- Nothing changed yet.
+- Improve test coverage show deprecation warnings when importing modules which
```
Comment (cjwatson) [inline CHANGES.rst:7]: "Improve test coverage show deprecation warnings ..." isn't quite grammatical English, and I'm not sure exactly what it's meant to say.

File: src/zope/testing/loghandler.py [hunk 2]
```diff
@@ -14,6 +14,12 @@
 """logging handler for tests that check logging output.
 """
 import logging
+import warnings
+
+
+warnings.warn(
+    'zope.testing.loghandler.Handler is deprecated. It probably'
+    'does not work on Python 3.', DeprecationWarning, stacklevel=2)
```
Comment (cjwatson) [inline src/zope/testing/loghandler.py:22]: There's a missing space on one or the other side of the line break here.

File: src/zope/testing/server.py [hunk 3]
```diff
@@ -25,13 +25,23 @@
...
+    from http.server import BaseHTTPRequestHandler, HTTPServer
+except ImportError:  # pragma: PY2
+    import urlparse
+    from BaseHTTPServer import BaseHTTPRequestHandler, HTTPServer
+
+
+def makeRequestHandler(http, user=None, password=None):  # pragma: PY2
+    warnings.warn(
+        'zope.testing.server.makeRequestHandler is deprecated. It probably'
+        'does not work on Python 3.', DeprecationWarning, stacklevel=2)
```
Comment (cjwatson) [inline src/zope/testing/server.py:43]: There's a missing space on one or the other side of the line break here.

File: src/zope/testing/server.py [hunk 4]
```diff
@@ -88,7 +98,10 @@ def addPortToURL(url, port):
     return url
 
 
-def startServer(http, url, user=None, password=None, port=8000):
+def startServer(http, url, user=None, password=None, port=8000):  # pragma: PY2
+    warnings.warn(
+        'zope.testing.server.startServer is deprecated. It probably'
+        'does not work on Python 3.', DeprecationWarning, stacklevel=2)
```
Comment (cjwatson) [inline src/zope/testing/server.py:104]: There's a missing space on one or the other side of the line break here.

**Review by cjwatson — APPROVED** [review]

File: CHANGES.rst (hunk 1 above)
Comment (icemac) [reply CHANGES.rst]: I split it into two bullet points, which make it hopefully clear what I am trying to say.

File: src/zope/testing/setupstack.txt [hunk 5]
```diff
@@ -228,6 +228,7 @@ Here's an example:
     ...
     ...     def setUp(self):
     ...         self.setUpDirectory()
+    ...         os.mkdir('example')
```
Comment (mgedmin) [inline src/zope/testing/setupstack.txt:231]: That is the point of this?

**Review by mgedmin — APPROVED** [review]
lgtm

File: src/zope/testing/setupstack.txt (hunk 5 above)
Comment (icemac) [reply src/zope/testing/setupstack.txt:231]: This is needed to get coverage for: https://github.com/zopefoundation/zope.testing/blob/8a46f1901bcaa871b919fc1ec2f22abd86587e94/src/zope/testing/setupstack.py#L66-L68

Comment (jugmac00) [reply src/zope/testing/setupstack.txt:231]: Maybe add a comment?

Comment (icemac) [reply src/zope/testing/setupstack.txt:233]: Done in https://github.com/zopefoundation/zope.testing/pull/38/commits/a6490fd98af64bbf1eb4706347782ccb98f025df

**Pull request merged**